docs/arch/02-core-concepts.md

```diff
@@ -0,0 +1,693 @@
+
+### Filter and Override
+
+**Filter and override** controls which tools are available to MCP clients and how they are presented.
+
+**Two complementary operations:**
+
+1. **Tool Filtering**: Whitelist specific tools by name
+   - Configured via `--tool` flags or `toolsFilter` config
+   - Tools not in filter list are hidden from clients
+   - Empty filter list means all tools are available
+
+2. **Tool Overriding**: Customize tool presentation
+   - Configured via `toolsOverride` map in config file
+   - Override tool names and/or descriptions
+   - Maps actual tool name to user-visible name/description
+
+**Two middlewares for consistency:**
+
+- **Tool Filter middleware**: Processes outgoing `tools/list` responses
+- **Tool Call Filter middleware**: Processes incoming `tools/call` requests
+
+Both middlewares share the same configuration to ensure clients only see tools they can call, and can only call tools they see.
+
+**Configuration:**
+- `toolsFilter` - List of allowed tool names (from `--tool` flags)
+- `toolsOverride` - Map from actual name to override (from config file)
+
+**Implementation:**
+- Middleware factories: `pkg/mcp/middleware.go`
+- Filter logic: `pkg/mcp/tool_filter.go`
+- Configuration: `pkg/runner/config.go`
+
+**Related concepts:** Middleware, Authorization
+
+### Authorize
+
+**Authorize** evaluates Cedar policies to determine if requests are permitted.
+
+**Authorization process:**
+1. Get parsed MCP data from context
+2. Get JWT claims from auth middleware
+3. Create Cedar entities (Principal, Action, Resource)
+4. Evaluate Cedar policies
+5. Allow or deny request
+
+**Policy language:**
+Cedar policies use:
+- `principal` - Who is making the request (from JWT)
+- `action` - What operation (from MCP method)
+- `resource` - What is being accessed (from MCP resource ID)
+
+**Example policy:**
+```cedar
+permit(
+  principal == Client::"user@example.com",
+  action == Action::call_tool,
+  resource == Tool::"web-search"
+);
+```
+
+**Implementation:**
+- Authz middleware: `pkg/authz/middleware.go`
+- Policy engine: Cedar (external library)
+
+**Related concepts:** Middleware, Authentication, Parse
+
+### Audit
+
+**Audit** logs MCP operations for compliance, monitoring, and debugging.
+
+**Audit events include:**
+- `mcp_tool_call` - Tool executions
+- `mcp_tools_list`, `mcp_resources_list`, `mcp_prompts_list` - List operations
+- `mcp_resource_read` - Resource access
+- `mcp_prompt_get` - Prompt retrieval
+- `mcp_initialize` - Session initialization
+- `sse_connection` - SSE connections
+- `http_request` - Generic HTTP requests
+- And others - see `pkg/audit/mcp_events.go` for complete list of 15 event types
+
+**Event data:**
+- Timestamp, source, outcome
+- Subjects (user, session)
+- Target (endpoint, method, resource)
+- Request/response data (configurable)
+- Duration and metadata
+
+**Implementation:**
+- Audit middleware: `pkg/audit/middleware.go`
+- Event types: `pkg/audit/events.go`
```

Comment:
File path reference is incorrect: `pkg/audit/events.go` does not exist.

Audit events are defined in:
- `pkg/audit/event.go` (core audit event structures)
- `pkg/audit/mcp_events.go` (MCP-specific event types)

Consider updating to reference both files.

Evidence: Verified via codebase exploration